File: packetCapture/src/main/java/com/b_knabe/packet_capture/core/util/common/FileManager.java
package com.b_knabe.packet_capture.core.util.common;

import java.io.File;

public class FileManager {
    private static void deleteFile(File file, boolean reserveSelf) {
        if (!file.exists()) return;
        if (file.isDirectory()) {
            File[] files = file.listFiles();
            for (File child : files) {
                delete(child);
            }
        }
        if (!reserveSelf) file.delete();
    }

    public static void delete(File file) {
        deleteFile(file, false);
    }

    public static void deleteUnder(File file) {
        deleteFile(file, true);
    }
}

File: packetCapture/src/main/java/com/ninpou/packetcapture/core/util/common/FileManager.kt
package com.ninpou.packetcapture.core.util.common

import java.io.File

object FileManager {
    private fun deleteFile(file: File, reserveSelf: Boolean) {
        if (!file.exists()) return
        if (file.isDirectory) {
            val files = file.listFiles()
            for (child in files) {
                delete(child)
            }
        }
        if (!reserveSelf) file.delete()
    }

    fun delete(file: File) {
        deleteFile(file, false)
    }

    @JvmStatic
    fun deleteUnder(file: File) {
        deleteFile(file, true)
    }
}